artiq/firmware/libproto_artiq/drtioaux_proto.rs

```diff
@@ -370,6 +416,47 @@ impl Packet {
                 writer.write_bool(succeeded)?;
                 writer.write_u8(retval)?;
             },
+            Packet::DmaAddTraceRequestWrite { id, last, length, trace } => {
+                writer.write_u8(0xb0)?;
+                writer.write_u32(id)?;
+                writer.write_bool(last)?;
+                // trace may be broken down to fit within drtio aux memory limit
+                // will be reconstructed by satellite
+                writer.write_bytes(&trace[0..length as usize])?;
```

Comment:
The hard array limitation was removed, so now it has to be kept in check by the writer.
I assume a sequential, send->ack->send->ack nature of the add packets, removing the need for keeping write offset, as the initial trace is constantly extended, until the last packet is received.